pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotInstanceRestletResource.java

```diff
@@ -196,4 +197,25 @@ public SuccessResponse updateInstance(
     }
     return new SuccessResponse("Instance successfully updated");
   }
+
+  @PUT
+  @Path("/instances/{instanceName}/updateTags")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Update the tags of the specified instance", consumes = MediaType.APPLICATION_JSON, notes = "Update the tags of the specified instance")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal error")})
+  public SuccessResponse updateInstanceTags(
+      @ApiParam(value = "Instance name", required = true, example = "Server_a.b.com_20000 | Broker_my.broker.com_30000") @PathParam("instanceName") String instanceName,
+      @ApiParam(value = "Comma separated tags list", required = true) @QueryParam("tags") String tags) {
+    LOGGER.info("Instance update request received for instance: {}", instanceName);
+    if (tags == null) {
+      throw new ControllerApplicationException(LOGGER, "Must provide tags to update", Response.Status.BAD_REQUEST);
+    }
+    PinotResourceManagerResponse response = pinotHelixResourceManager.updateInstanceTags(instanceName, tags);
+    if (!response.isSuccessful()) {
+      throw new ControllerApplicationException(LOGGER,
+          "Failure to update instance tags. Reason: " + response.getMessage(), Response.Status.INTERNAL_SERVER_ERROR);
```

Comment:
If the instance does not exist, we should return `BAD_REQUEST`

Reply:
Instance check is happening in PinotHelixResourceManager. How do i return BAD_REQUEST from PinotHelixResourceManager? I want that instance check to remain in PinotHelixResourceManager, because that method is synchronized and will protect against race condition with delete.